src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodec.java

```diff
@@ -106,30 +137,79 @@ public interface FromJava<T, R> {
      * <p>Unsafe because it does not catch any errors from the conversion, because there are none.
      *
      * @return
-     * @param <T>
+     * @param <JavaT>
      */
-    static <T> FromJava<T, T> unsafeIdentity() {
-      return (t, targetType) -> t;
+    static <JavaT> ToCQL<JavaT, JavaT> unsafeIdentity() {
+      return (toCQLType, value) -> value;
     }
 
     /**
      * Returns an instance that converts the value to the target type, catching any arithmetic
-     * exceptions and throwing them as a {@link FromJavaCodecException}
+     * exceptions and throwing them as a {@link ToCQLCodecException}
      *
      * @param function the function that does the conversion, it is expected it may throw a {@link
      *     ArithmeticException}
      * @return
-     * @param <T>
-     * @param <R>
+     * @param <JavaT>
+     * @param <CqlT>
      */
-    static <T extends Number, R> FromJava<T, R> safeNumber(Function<T, R> function) {
-      return (t, targetType) -> {
+    static <JavaT extends Number, CqlT> ToCQL<JavaT, CqlT> safeNumber(
+        Function<JavaT, CqlT> function) {
+      return (toCQLType, value) -> {
         try {
-          return function.apply(t);
+          return function.apply(value);
         } catch (ArithmeticException e) {
-          throw new FromJavaCodecException(t, targetType, e);
+          throw new ToCQLCodecException(value, toCQLType, e);
         }
       };
     }
   }
+
+  /**
+   * Function interface that is used by the codec to convert value returned by CQL into a {@link
+   * JsonNode} that can be used to construct the response document for a row.
+   *
+   * <p>The interface is used so the conversation function can throw the checked {@link
+   * ToJSONCodecException}, it is also given the CQL data type to make better exceptions.
+   *
+   * <p>Use the static constructors on the interface to get instances, see it's use in the {@link
+   * JSONCodecRegistry}
+   *
+   * @param <CqlT> The type Java object the CQL driver expects
+   */
+  @FunctionalInterface
+  public interface ToJSON<CqlT> {
+
+    /**
+     * Converts the value read from CQL to a {@link JsonNode}
+     *
+     * @param objectMapper A {@link ObjectMapper} to use to create the {@link JsonNode} if needed.
+     * @param fromCQLType The CQL {@link DataType} of the column that was read from CQL.
+     * @param value The value that was read from the CQL driver.
+     * @return A {@link JsonNode} that represents the value, this is just the value does not include
+     *     the column name.
+     * @throws ToJSONCodecException Checked exception raised for any error, users of the function
+     *     must catch and convert to the appropriate error for the use case.
+     */
+    JsonNode apply(ObjectMapper objectMapper, DataType fromCQLType, CqlT value)
```

Comment:
Since we don't have to use very vague name like "apply()", could use a more meaningful name;
`toJson()` or whatever.

Reply:
fixed as toJson